tsdb/meta.go

```diff
@@ -833,10 +833,30 @@ func (m *Measurement) idsForExpr(n *influxql.BinaryExpr) (SeriesIDs, influxql.Ex
 	return nil, nil, nil
 }
 
+// FilterExprs represents a map of series IDs to filter expressions.
+type FilterExprs map[uint64]influxql.Expr
+
+// DeleteBoolLiteralTrues deletes all elements whose filter expression is a boolean literal true.
+func (fe FilterExprs) DeleteBoolLiteralTrues() {
+	for id, expr := range fe {
+		if e, ok := expr.(*influxql.BooleanLiteral); ok && e.Val == true {
+			delete(fe, id)
```

Comment:
It's safe.  From the golang spec...
```
The iteration order over maps is not specified and is not guaranteed to be the same from one
iteration to the next. If map entries that have not yet been reached are removed during iteration,
the corresponding iteration values will not be produced. If map entries are created during
iteration, that entry may be produced during the iteration or may be skipped.
```
https://golang.org/ref/spec#For_statements